figures/plot_benchmark_bilevel.py

```diff
@@ -129,14 +133,22 @@
     ] = 'soba full batch'
 
     # Select curve that reach the lowest point
-    # import ipdb; ipdb.set_trace()
-    to_plot = (
-        df.query('stop_val <= 100')
-        .groupby(['solver', 'solver_name', 'stop_val'])
-        .median()
-        .reset_index().sort_values(metric)
-        .groupby('solver').first()['solver_name']
-    )
+    if args.criterion == '100':
```

Comment:
avoid duplicating all code just to add a filer. just add a filter that is always true if criterion is not 0, else 100?